soft_bfddpd/bfddpd.c

```diff
@@ -139,30 +133,33 @@ parse_address(const char *arg, struct sockaddr *sa, socklen_t *salen)
 	struct sockaddr_in *sin;
 	struct sockaddr_in6 *sin6;
 	struct sockaddr_un *sun;
-	char *sptr;
-	size_t typelen;
+	char *sptr, *saux;
+	size_t slen;
 	char type[64];
 	char addr[64];
 
+	/* Basic parsing: find ':' to figure out type part and address part. */
 	sptr = strchr(arg, ':');
 	if (sptr == NULL) {
 		fprintf(stderr, "Invalid address format: %s\n", arg);
 		exit(1);
 	}
 
 	/* Calculate type string size. */
-	typelen = (size_t)(sptr - arg);
+	slen = (size_t)(sptr - arg);
 
 	/* Copy type string. */
 	sptr++;
-	memcpy(type, arg, typelen);
-	type[typelen] = 0;
+	memcpy(type, arg, slen);
```

Comment:
Consider guarding against overwriting the boundary of type should slen be greater than 64.